### PR TITLE
Add translations article

### PR DESCRIPTION
**Pull request by fritzmg**

### CODE REVIEW

**Comment by Toflar** [comment]
Great work! What I'm missing is a note about the general way of using translations in Symfony. You can use the regular Symfony translator but for legacy reasons there's also the translation handling of Contao. Something like that, do you know what I mean? Some more general info to answer the question of "why the heck do I even have to use the Contao translation stuff?".

**Comment by fritzmg** [comment]
Something like this: 242da8f74b8a9669430cb8b225e2c3cfcacfc3e6 ?

File: docs/dev/framework/translations.md [hunk 1]
```diff
@@ -6,4 +6,246 @@ aliases:
...
+| `CONFIRM`  | `default`   |                                            |
+| `DP`       | `default`   | Date picker.                               |
+| `COLS`     | `default`   | Layout section names.                      |
+| `SECTIONS` | `default`   | Layout section positions.                  |
+| `DCA`      | `default`   | Various data container view translations.  |
+| `XPT`      | `exception` | Error messages for the front and back end. |
+| `XPL`      | `explain`   | Help wizard content.                       |
+| `LNG`      | `languages` | Translation of language names.             |
+| `MOD`      | `modules`   | Back end module labels and descriptions.   |
+| `FMD`      | `moduels`   | Front end module labels and descriptions.  |
```
Comment (xchs) [inline docs/dev/framework/translations.md]:
```suggestion
| `FMD`      | `modules`   | Front end module labels and descriptions.  |
```

File: docs/dev/framework/translations.md [hunk 2]
```diff
@@ -6,4 +6,246 @@ aliases:
...
+      <trans-unit id="tl_content.text.1">
+        <target>Geben Sie hier den Text-Inhalt ein.</target>
+      </trans-unit>
+    </body>
+  </file>
+</xliff>
+```
+
+When extending translations, only the choice of the translation domain is relevant. 
+Categories and keys for for new translations can be chosen at your own discretion.
```
Comment (xchs) [inline docs/dev/framework/translations.md]:
```suggestion
Categories and keys for new translations can be chosen at your own discretion. 
```

**Review by xchs — CHANGES_REQUESTED** [review]
LGTM! 👍

**Review by xchs — APPROVED** [review]

**Review by Toflar — APPROVED** [review]
Great work!

**Pull request merged**